sktime/forecasting/compose/_reduce.py

```
@@ -2433,35 +2479,37 @@ def _fit(self, y, X, fh):
         # todo: very similar to _fit_concurrent of DirectReductionForecaster - refactor?
         from sktime.transformations.series.lag import Lag
 
-        impute_method = self._impute_method
+        # impute_method = self._impute_method
+        X_treatment = self.X_treatment
 
-        # lagger_y_to_X_ will lag y to obtain the sklearn X
+        # lagger_y_to_X_ will lag y and later concat X to obtain the sklearn X
         lags = self._lags
         lagger_y_to_X = Lag(lags=lags, index_out="extend")
 
-        if impute_method is not None:
-            lagger_y_to_X = lagger_y_to_X * impute_method.clone()
+        # if impute_method is not None:
+        #    lagger_y_to_X = lagger_y_to_X * impute_method.clone()
         self.lagger_y_to_X_ = lagger_y_to_X
 
         Xt = lagger_y_to_X.fit_transform(y)
 
         # lag is 1, since we want to do recursive forecasting with 1 step ahead
-        lag_plus = Lag(lags=1, index_out="extend")
+        # column names will be kept for consistency
+        lag_plus = Lag(lags=1, index_out="extend", keep_column_names=True)
         Xtt = lag_plus.fit_transform(Xt)
         Xtt_notna_idx = _get_notna_idx(Xtt)
         notna_idx = Xtt_notna_idx.intersection(y.index)
 
+        # yt is the target forecast value
         yt = y.loc[notna_idx]
         Xtt = Xtt.loc[notna_idx]
 
-        # we now check whether the set of full lags is empty
-        # if yes, we set a flag, since we cannot fit the reducer
-        # instead, later, we return a dummy prediction
         if len(notna_idx) == 0:
             self.estimator_ = y.mean()
```

This line currently creates a bug. y.mean() returns a pandas object, not an estimator. 
In addition, I believe an error should be raised here, instead of returning a dummy prediction. Users could use FallbackForecaster then to choose the default behaviour in such cases.

---

question, what was previous behaviour?

I think the previous behaviour was to track this case and fill the mean.
This allowed to treat edge cases that arise in important common componsites, like very small windows in a windowed tuning composites, without problems. Removing the treatment of this edge case would cause failures in common composites.